### PR TITLE
feature/identity-service

### PR DESCRIPTION
**Pull request by Mohammed-Alhams**

### CODE REVIEW

File: service_identity/src/main/kotlin/org/thechance/service_identity/data/geteway/DataBaseGatewayImp.kt [hunk 1]
```diff
@@ -1,85 +1,260 @@
...
-                PermissionCollection::id eq ObjectId(permissionId),
+                PermissionCollection::id eq permissionId,
                 PermissionCollection::isDeleted eq false
             ),
             update = setValue(PermissionCollection::isDeleted, true)
         ).isUpdatedSuccessfully()
     }
 
-    override suspend fun updatePermission(permissionId: String, permission: Permission): Boolean {
+    override suspend fun getListOfPermission(permissionId: Int): List<Permission> {
```
Comment (aya-seif) [inline service_identity/src/main/kotlin/org/thechance/service_identity/data/geteway/DataBaseGatewayImp.kt:113]: why not getPermissions ?

File: service_identity/src/main/kotlin/org/thechance/service_identity/data/mappers/DetailedUser.kt [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
-fun DetailedUserDto.toEntity(): User {
+fun DetailedUserDto.toEntity(wallet: Wallet): User {
     return User(
         id = id,
         fullName = fullName,
         username = username,
         password = password,
-        email = email,
-        walletId = walletId,
+        email = "",
```
Comment (aya-seif) [inline service_identity/src/main/kotlin/org/thechance/service_identity/data/mappers/DetailedUser.kt:31]: why didn't you make email nullable?

Comment (Mohammed-Alhams) [reply service_identity/src/main/kotlin/org/thechance/service_identity/data/mappers/DetailedUser.kt:31]: We will delete it anyway

File: service_identity/src/main/kotlin/org/thechance/service_identity/data/geteway/DataBaseGatewayImp.kt (hunk 1 above)
Comment (Mohammed-Alhams) [reply service_identity/src/main/kotlin/org/thechance/service_identity/data/geteway/DataBaseGatewayImp.kt:113]: Ok, we will refractor it to the suggested name

File: service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/user_management/UserManagementUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    suspend fun addPermissionToUser(userId: String, permissionId: Int): Boolean
+
+    suspend fun removePermissionFromUser(userId: String, permissionId: Int): Boolean
+
+    suspend fun getUserPermissions(userId: String): List<Permission>
+
+    suspend fun getUserById(id: String): User
+
+    suspend fun getUsersList(fullName: String, username: String): List<User>
```
Comment (aya-seif) [inline service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/user_management/UserManagementUseCase.kt:16]: what about getUsers ?

**Review by aya-seif — APPROVED** [review]

**Review by devsadeq — APPROVED** [review]

**Pull request merged**